### PR TITLE
Bridge 1486 data groups

### PR DESCRIPTION
**Pull request by syoung-smallwisdom**
Add support for data groups tracking and clean up the sample project.

### CODE REVIEW

File: BridgeAppSDK/SBADataGroupsStep.swift [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    public func stepResult(currentGroups: [String]?) -> ORKStepResult? {
+        
+        // Look for a current choice from the input groups
+        let currentChoices: [Any]? = {
+            // Check that the current group is non-nil
+            guard currentGroups != nil, let answerFormat = self.formItems?.first?.answerFormat as? ORKTextChoiceAnswerFormat
+            else {
+                return nil
+            }
+            // Create the itersection set that is the values from the current group that are in this steps subset of data groups
```
Comment (Erin-Mounts) [inline BridgeAppSDK/SBADataGroupsStep.swift]: "itersection" --> "intersection"

File: BridgeAppSDK/SBADataGroupsStep.swift [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+        
+        // If found, then create a questionResult for that choice
+        let questionResult = ORKChoiceQuestionResult(identifier: self.identifier)
+        questionResult.choiceAnswers = currentChoices
+        return ORKStepResult(stepIdentifier: self.identifier, results: [questionResult])
+    }
+    
+    /**
+     Union the input data groups with the given step result.
+     @param  dataGroups     The current data groups set
```
Comment (Erin-Mounts) [inline BridgeAppSDK/SBADataGroupsStep.swift]: should the param name be previousGroups?

File: BridgeAppSDK/SBADataGroupsStep.swift [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+     @param  stepResult     The step result to use to get the new selection
+     @return                The union set of data groups based on the current and selected
+     */
+    public func union(previousGroups: Set<String>?, stepResult: ORKStepResult) -> Set<String> {
```
Comment (Erin-Mounts) [inline BridgeAppSDK/SBADataGroupsStep.swift:116]: From above it looks like the stepResult consists of the intersection of the user's responses with the current data groups, and then from the doc comments it sounds like this function just unions those with the current data groups, which would always just end up returning the current data groups. Most likely, I'm just misunderstanding what's going on here...?

File: BridgeAppSDK/SBATaskViewController.swift [hunk 4]
```diff
@@ -141,15 +156,35 @@ open class SBATaskViewController: ORKTaskViewController, ORKTaskViewControllerDe
     
     // MARK: ORKTaskResultSource
     
-    // Look to the task attached to this view controller and return that as the result source
-    // If there is none set as the override.
+    // syoung 09/30/2016 Override the result source so that this task view controller can forward results
+    // from either the task or an external result source. This wacky
```
Comment (Erin-Mounts) [inline BridgeAppSDK/SBATaskViewController.swift]: This wacky what? ;)

File: BridgeAppSDK/SBATrackedDataObjectCollection.swift [hunk 5]
```diff
@@ -232,4 +232,11 @@ extension SBATrackedDataObjectCollection: SBABridgeTask, SBAStepTransformer, SBA
         return selectedItems.find({ $0.tracking }) != nil
     }
 
+    open func commitChanges(user: SBAUserWrapper, task: SBANavigableOrderedTask, taskResult: ORKTaskResult) {
+
```
Comment (Erin-Mounts) [inline BridgeAppSDK/SBATrackedDataObjectCollection.swift]: Is this meant to be overridden in subclasses? Implementing a protocol method as a stub? Something else? A quick comment would be helpful here

**Review by Erin-Mounts — COMMENTED** [review]
Mostly nits about comments or lack thereof, couple of questions

File: BridgeAppSDK/SBADataGroupsStep.swift (hunk 1 above)
Comment (syoung-smallwisdom) [reply BridgeAppSDK/SBADataGroupsStep.swift]: fixed

File: BridgeAppSDK/SBADataGroupsStep.swift (hunk 2 above)
Comment (syoung-smallwisdom) [reply BridgeAppSDK/SBADataGroupsStep.swift]: fixed

File: BridgeAppSDK/SBADataGroupsStep.swift (hunk 3 above)
Comment (syoung-smallwisdom) [reply BridgeAppSDK/SBADataGroupsStep.swift:116]: Comment has been expanded.

File: BridgeAppSDK/SBATaskViewController.swift (hunk 4 above)
Comment (syoung-smallwisdom) [reply BridgeAppSDK/SBATaskViewController.swift]: oops. fixed.

File: BridgeAppSDK/SBATrackedDataObjectCollection.swift (hunk 5 above)
Comment (syoung-smallwisdom) [reply BridgeAppSDK/SBATrackedDataObjectCollection.swift]: removed. this was dead code.

**Comment by Erin-Mounts** [comment]
looks good

**Pull request merged**